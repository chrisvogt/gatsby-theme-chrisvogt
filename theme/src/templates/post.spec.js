import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'

import Post, { Head } from './post'

const data = {
  mdx: {
    fields: {
      category: 'Mock Category',
    },
    frontmatter: {
      date: 'Mon, 17 Jun 2024 03:30:26 GMT',
      title: 'A Mock Blog Post',
      description: 'This is a mock description',
      banner: 'mock-banner.jpg',
    }
  }
}

jest.mock('gatsby');
jest.mock('../components/layout', () => {
  return ({children}) => <div className='layoutMock'>{children}</div>
});

jest.mock('../components/seo', () => {
  return ({children}) => <div className='seoMock'>{children}</div>
});

const BlogPostContent = <div>Lorum ipsum dolor sit amet.</div>;

describe('Blog Post', () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => data)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the snapshot', () => {
    const tree = renderer.create(<Post data={ data }>{ BlogPostContent }</Post>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not render category when not provided', () => {
    const noCategoryData = {
      ...data,
      mdx: {
        ...data.mdx,
        fields: {
          category: null
        }
      }
    }
    const tree = renderer.create(<Post data={ noCategoryData }>{ BlogPostContent }</Post>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Seo component with the correct props', () => {
    const seoTree = renderer.create(<Head data={ data } />).toJSON()
    expect(seoTree).toMatchSnapshot()
  })
})
