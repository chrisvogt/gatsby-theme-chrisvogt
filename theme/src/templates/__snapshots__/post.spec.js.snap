// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post matches the snapshot 1`] = `
<div
  className="layoutMock"
>
  <div
    className="css-43plg"
  >
    <div
      className="css-1qfp07i"
    >
      <article
        className="h-entry c1v0-blog-post"
      >
        <div
          className="p-category css-d5b7h2"
        >
          Mock Category
        </div>
        <h1
          className="p-name css-12fp046"
        />
        <time
          className="dt-published created"
        >
          Published 
          Mon, 17 Jun 2024 03:30:26 GMT
        </time>
        <div
          className="e-content article-content"
        >
          <div>
            Lorum ipsum dolor sit amet.
          </div>
        </div>
      </article>
    </div>
  </div>
</div>
`;
